### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by melgor** [pranv/ARC#1]
Great work u/pranv! I was thinking about similar approch, but I have problem to code it correctly (I was trying to make attention of both images at once).
I have some question:

- So you use a hard-attention rather than soft-attention. Any reason?

- In paper you mention that:
`We arrived at the iterative cycling paradigm after trying out many approaches to attend to multiple images at once on a few toy datasets`
Could you list method which fails to learn?

- At repo I see the code related to LFW. Do you have any results on LFW Verification protocol (using net learned on CASIA)?

- I see that for Face Verification you used images 32x32. As I understand, the bigger images are too computational intensive? How long does it take to learn model on CASIA?

**Comment by pranv** [comment]
> So you use a hard-attention rather than soft-attention. Any reason?

Soft attention can be trained by backpropagation. It is easier train and generally trains faster than attention mechanisms that have to be trained with Reinforcement Learning. But that said, hard attention does have a lower computational cost during inference. It is an interesting direction to explore.

> Could you list method which fails to learn?

I tried the thing you mentioned: attending simultaneously to 2 images at each time step, but this did not work well. A hypothesis that we formed was that the correlation structure present in the vector with 2 attention glimpses makes it hard for the network to learn (but this was not properly validated). I tried a few more ways like having 2 separate RNNs that take each other's hidden state as input at each time step, or, having some dense layers on top of attention before giving it to the RNN controller - but these did not work or didn't work as good as the one we finally settled on in the paper

> At repo I see the code related to LFW. Do you have any results on LFW Verification protocol (using net learned on CASIA)?

My initial aim was to train a network on LFW for Face Recognition. But LFW was itself too small to train ARCs on without over fitting badly (even ConvNets overfit quickly). So we decided to use CASIA instead which has a lot more face samples (it was a pain to get this dataset). It would have been ideal if we could train on CASIA and report results using the standard LFW testing setup. But I don't think we have time for that. I only did the Omniglot side of things. My second author Shubham worked on CASIA and ImageNet and now even he's busy with some other things. So sorry about that. I hope someday I or someone else can work on it. 

> I see that for Face Verification you used images 32x32. As I understand, the bigger images are too computational intensive? How long does it take to learn model on CASIA?

We actually now have results using 64x64 images and we will update the paper soon. Shubham says that it took 3 days to train it but the accuracy was better at 84.61%  test set (versus 81.73% using 32x32 images)

Feel free to ask any and all questions you have on our paper, in addition to any feedback and/or criticism. I will try my best to address them.
